### ISSUE TITLE
[BUG] reinstall 1st harvester node cannot join back to original cluster

### ISSUE THREAD

**Issue by tjjh89017** [rancher/rancher#38640]
**Rancher Server Setup**
- Rancher version: v2.6.4-harvester3
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): rke2 1.22.12+rke2r1, Rancherd v0.0.1-alpha13-harvester1 (https://github.com/harvester/rancherd)
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.22
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:

**Describe the bug**

Install 3 harvester node and promote 3 node to control-plane.
Shutdown Node1 (first master node) and remove Node1 via WebUI
Reinstall Node1 with same network config and join back to existing cluster (Node2, Node3)
Node1 will stuck in bootstrap status and didn't join cluster

**To Reproduce**

Install 3 harvester node and promote 3 node to control-plane.
Shutdown Node1 (first master node) and remove Node1 via WebUI
Reinstall Node1 with same network config and join back to existing cluster (Node2, Node3)
Node1 will stuck in bootstrap status and didn't join cluster

**Result**

**Expected Result**

Join back to cluster

**Screenshots**

**Additional context**

secret custom-<New Node1>-machine-bootstrap will be always existing.
secret custom-<New Node1>-machine-plan will not have any data.
rancher-system-agent will keep waiting for custom-<New Node1>-machine-plan

related issue: https://github.com/harvester/harvester/issues/2665

**Comment by tjjh89017** [comment]
before reinstall 1st node
[supportbundle_074224c6-2749-472b-baf1-d487e223628d_2022-08-25T03-21-33Z-before.zip](https://github.com/rancher/rancher/files/9421114/supportbundle_074224c6-2749-472b-baf1-d487e223628d_2022-08-25T03-21-33Z-before.zip)

**Comment by tjjh89017** [comment]
After
[supportbundle_074224c6-2749-472b-baf1-d487e223628d_2022-08-25T03-36-40Z.zip](https://github.com/rancher/rancher/files/9421142/supportbundle_074224c6-2749-472b-baf1-d487e223628d_2022-08-25T03-36-40Z.zip)

**Comment by tjjh89017** [comment]
```
node1:/home/rancher # journalctl -u rancherd | cat
-- Logs begin at Thu 2022-08-25 03:33:48 UTC, end at Thu 2022-08-25 03:38:44 UTC. --
Aug 25 03:34:08 node1 systemd[1]: Starting Rancher Bootstrap...
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Loading config file [/usr/share/rancher/rancherd/config.yaml.d/50-defaults.yaml]"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Loading config file [/usr/share/rancher/rancherd/config.yaml.d/91-harvester-bootstrap-repo.yaml]"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Loading config file [/oem/99_custom.yaml]"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Loading config file [/etc/rancher/rancherd/config.yaml]"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Bootstrapping Rancher (v2.6.4-harvester3/v1.22.12+rke2r1)"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Writing plan file to /var/lib/rancher/rancherd/plan/plan.json"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Applying plan with checksum "
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="No image provided, creating empty working directory /var/lib/rancher/rancherd/plan/work/20220825-033408-applied.plan/_0"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="Running command: /usr/bin/env [sh /var/lib/rancher/rancherd/install.sh]"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="[stdout]: [INFO]  Using default agent configuration directory /etc/rancher/agent"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="[stdout]: [INFO]  Using default agent var directory /var/lib/rancher/agent"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="[stderr]: [WARN]  /usr/local is read-only or a mount point; installing to /opt/rancher-system-agent"
Aug 25 03:34:08 node1 rancherd[1892]: time="2022-08-25T03:34:08Z" level=info msg="[stdout]: [INFO]  Determined CA is necessary to connect to Rancher"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Successfully downloaded CA certificate"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Value from https://10.0.0.228:443/cacerts is an x509 certificate"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Successfully tested Rancher connection"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Downloading rancher-system-agent from https://10.0.0.228:443/assets/rancher-system-agent-amd64"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Successfully downloaded the rancher-system-agent binary."
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Generating Cattle ID"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Successfully downloaded Rancher connection information"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  systemd: Creating service file"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Creating environment file /etc/systemd/system/rancher-system-agent.env"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Enabling rancher-system-agent.service"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stderr]: Created symlink /etc/systemd/system/multi-user.target.wants/rancher-system-agent.service → /etc/systemd/system/rancher-system-agent.service."
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stdout]: [INFO]  Starting/restarting rancher-system-agent.service"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="No image provided, creating empty working directory /var/lib/rancher/rancherd/plan/work/20220825-033408-applied.plan/_1"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="Running command: /usr/bin/rancherd [probe]"
Aug 25 03:34:09 node1 rancherd[1892]: time="2022-08-25T03:34:09Z" level=info msg="[stderr]: time=\"2022-08-25T03:34:09Z\" level=info msg=\"Running probes defined in /var/lib/rancher/rancherd/plan/plan.json\""
Aug 25 03:34:10 node1 rancherd[1892]: time="2022-08-25T03:34:10Z" level=info msg="[stderr]: time=\"2022-08-25T03:34:10Z\" level=info msg=\"Probe [kubelet] is unhealthy\""

```

**Comment by Oats87** [comment]
@tjjh89017 

I'm unfamiliar with the inner workings of Harvester here. Was a fix for https://github.com/rancher/rancher/issues/37462 implemented? 

I made a code change in Rancher that only exists in v2.6.7 or newer of Rancher that was a bug fix. This bug fix, in addition to a change in behavior in rancherd, is required to fix this type of issue.

I think this might be part of https://github.com/harvester/harvester/issues/2191 and essentially a duplicate of 37462, but I will wait for you to look at both before closing/consolidating the issues.

**Comment by tjjh89017** [comment]
@Oats87 It seems it's similar to 37462
It will check it
thanks for that info

**Comment by tjjh89017** [comment]
@Oats87 There is a big different between this one and 37462.
We cannot join cluster.
but 37462 is fail in promotion.
I will grab some log of rke2-server & rke2-agent to you

**Comment by tjjh89017** [comment]
after setting new init-node label, it will waiting for cluster agent to connect.
but node1 still can't join and stuck.

and I could rancherd require at least one node with "role: cluster-init".
but because we remove the first node.
so there is no one with "cluster-init"

**Comment by tjjh89017** [comment]
after testing the workaround https://github.com/harvester/harvester/issues/2191#issuecomment-1115794201 .
It didn't work.
and I found secret "custom-XXXXX-machine-plan" is empty
and `rancher-system-agent` listen secret and got empty secret.

**Comment by bk201** [comment]
This is already fixed in Rancher v2.6.9.
